src/imap.rs

```diff
@@ -1611,6 +1622,50 @@ impl Imap {
 
         Ok((last_uid, received_msgs))
     }
+
+    /// Retrieves server metadata if it is supported.
+    ///
+    /// We get [`/shared/comment`](https://www.rfc-editor.org/rfc/rfc5464#section-6.2.1)
+    /// and [`/shared/admin`](https://www.rfc-editor.org/rfc/rfc5464#section-6.2.2)
+    /// metadata.
+    pub(crate) async fn get_metadata(&mut self, context: &Context) -> Result<()> {
```

Comment:
Maybe a better name would be `fetch_metadata()` or so because it doesn't return anything, but it's ok as is